model-optimizer/mo/utils/cli_parser_test.py

```diff
@@ -355,6 +355,15 @@ def test_mean_do_not_match_input(self):
         mean_values = parse_tuple_pairs("input_not_present(255),input2(255)")
         self.assertRaises(Error, get_mean_scale_dictionary, mean_values, scale_values, "input1,input2")
 
+    def test_values_match_input_name(self):
+        # before fix cli_parser was confused when input had a substring that matches scale/mean values
+        values = parse_tuple_pairs("input255(255),input255.0(255.0)")
+        exp_res = {'input255': 255.0, 'input255.0': 255.0}
+        self.assertEqual(exp_res, values)
```

Comment:
I would also want you to cover cases:
1. square brackets - input255.0[1.0,2.0,3.0];
2. several dots - input255.0.3[2.0,3.0];
3. several inputs - input.1[2.0,3.0],input.2[4.0,5.0],;

Reply:
@rkazants, thanks for your feedback!
1. Luckily, square brackets are covered in test_multi_mean_scale_arrays_no_input ("[1.1,22.22,333.333],[2.1,33.22,444.333]"); test_multi_mean_scale_input ("data[1.1,22.22,333.333],info[2.1,33.22,444.333]")
2. algorithm is not bound to dots outside of brackets so inputs with such names are safely processed. Added one test case to reduce change that someone in future will break it
3. added input with several dots in name to the same test so now there is 3 inputs but this does not change anything in principle -- 2 inputs are processed same as 4, 5 and etc.

Also added some code for deeper test result comparison. Please review.